src/components/NavModal.jsx

```diff
@@ -0,0 +1,100 @@
+import { Transition } from "@headlessui/react";
+import { Fragment } from "react";
+
+const NavModal = ({ isShowing, setIsShowing }) => {
+
+  const closeModal = () => {
+    setIsShowing(!isShowing);
+    document.body.classList.remove('overflow-hidden');
```

Comment:
I think this makes sense to move up a level. closeModal should do this logic and pass the function down.